### ISSUE TITLE
Presentation using two different applications

### ISSUE THREAD

**Issue by decentralgabe** [openid/OpenID4VP#60]
At TBD we have a setup that looks like this:
- An identity wallet (holds keys, identities, credentials)
- Applications that leverage data the identity wallet has (like a financial app)

A common flow would be:

1. User uses their **Financial App** to initiate a presentation flow (need 1 or more VCs) 
2. User presents a set of VCs as prerequisites from one or more **Identity Wallets** (e.g. device wallet, other identity wallet)
3. User needs to go back to their **Financial App** after presentation to continue with their flow (e.g. opening a bank account)

Put simply: the application a user initiates the presentation flow with is different from the application(s) that will present the credential(s).

Can this be supported today with OID4VP?

**Comment by paulbastian** [comment]
Is the question about whether the requesting application can be non-browser / native app and you want to be redirected in the requesting app?

**Comment by decentralgabe** [comment]
@paulbastian the redirect does not need to happen in the requesting app, it would be nice if the protocol understood that there are multiple applications in play. Specifically, the presenting app is not the same as the initiating app.

**Comment by paulbastian** [comment]
I think that's not a problem.

You can go
- browser - wallet app - back to browser
- verifier app - wallet app - back to verifier app 

The first jump happens through the various wallet invocation methods like custom Uri scheme, the second with a redirect

**Comment by decentralgabe** [comment]
thanks! (cc: @csuwildcat) for confirmation that this is sufficient

**Comment by csuwildcat** [comment]
So just to clarify, @paulbastian, this means I can:

1. Be on Cool App
2. Cool App triggers a flow where Verifier Entity needs credentials
3. Cool App relays/hands off the invocation to Wallet
4. Wallet creates a presentation and sends to Verifier (or passes the presentation back to Cool App for passage to Verifier)
5. Wallet returns the user to Cool App

?

**Comment by jogu** [comment]
Yeah, I would expect that flow to work. Which step are you worried won't work?

**Comment by jogu** [comment]
Although I'm not sure I really understand what's going on in step 2. What kind of flow is that, is the "verifier entity" an sdk embedded in the app or a website in an in-app browser tab or something else?

**Comment by decentralgabe** [comment]
@jogu the verifier entity is a server somewhere, accessed via Cool App (some backend calls)

**Comment by jogu** [comment]
I think that should work fine. I assume there's some level to inherent trust between cool app & the verifier backend, as the VP protocol will treat them as a single entity.  There's some questions about which app the client_id would technically belong to, but so long as there's a redirect URL registered to the client_id that coolapp can claim as a deep link it'll work.
